dags/iop/parser.py

```diff
@@ -100,3 +111,50 @@ def _get_extracted_arxiv_eprint_value(self, article):
             self.logger.error("The arXiv value is not valid.", dois=self.dois)
         except AttributeError:
             self.logger.error("No arXiv eprints found", dois=self.dois)
+
+    def _get_date_element(self, article):
+        try:
+            date = (
+                article.find("front/article-meta/history/date[@date-type='published']")
+                or article.find("front/article-meta/history/date[@date-type='epub']")
+                or article.find("front/article-meta/history/pub-date[@pub-type='ppub']")
+                or article.find("front/article-meta/pub-date")
+            )
+            return date
+        except AttributeError:
+            return
+
+    def _get_date_published(self, article):
+        date_element = self._get_date_element(article)
+        return self._get_date(date_element)
+
+    def _get_day(self, date):
+        try:
+            day = int(date.find("day").text)
+            return day
+        except (AttributeError, TypeError):
+            self.logger.error("Cannot find day of date_published in XML")
+
+    def _get_month(self, date):
+        try:
+            month = int(date.find("month").text)
+            return month
+        except (AttributeError, TypeError):
+            self.logger.error("Cannot find month of date_published in XML")
+
+    def _get_year(self, date):
+        try:
+            year = int(date.find("year").text)
+            return year
+        except (AttributeError, TypeError):
+            self.logger.error("Cannot find year of date_published in XML")
+
+    def _get_date(self, date):
+        try:
+            return datetime.date(
+                day=self._get_day(date),
```

Comment:
are you sure that you don't want to return date if there's no `day`?

Reply:
If the date is corrupted, I am not sure what to do:
1) if there is just one of the variables? (day || month || year)
2) if there are just two of them?